### PR TITLE
feat: 🎸 add `getInvolvedParties` to ConfidentialTransaction

### PR DESCRIPTION
**Pull request by polymath-eric**
### Description

add method to fetch involved parties in a transaction

### Breaking Changes

none

### JIRA Link

✅ Closes: [DA-991](https://polymesh.atlassian.net/browse/DA-991)

### Checklist

- [ ] Updated the Readme.md (if required) ?

[DA-991]: https://polymesh.atlassian.net/browse/DA-991?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/api/entities/confidential/__tests__/ConfidentialTransaction/index.ts [hunk 1]
```diff
@@ -149,6 +150,47 @@ describe('ConfidentialTransaction class', () => {
...
+
+      expect(result).toEqual(
+        expect.arrayContaining([
+          expect.objectContaining({ did: '0x01' }),
+          expect.objectContaining({ did: '0x02' }),
+        ])
+      );
+    });
+
+    it('should throw an error if no parties are found', async () => {
```
Comment (prashantasdeveloper) [inline src/api/entities/confidential/__tests__/ConfidentialTransaction/index.ts:179]:
```suggestion
    it('should throw an error if no parties are found', () => {
```

File: src/api/entities/confidential/__tests__/ConfidentialTransaction/index.ts [hunk 2]
```diff
@@ -149,6 +150,47 @@ describe('ConfidentialTransaction class', () => {
...
+      dsMockUtils.createQueryMock('confidentialAsset', 'transactionParties', {
+        entries: [],
+      });
+
+      const expectedError = new PolymeshError({
+        code: ErrorCode.DataUnavailable,
+        message:
+          'No involved parties were found for this transaction. Its likely been completed and the chain storage has been pruned',
+      });
```
Comment (prashantasdeveloper) [inline src/api/entities/confidential/__tests__/ConfidentialTransaction/index.ts:188]: I thought this should fail as we didn't add the `data` part here

**Review by prashantasdeveloper — CHANGES_REQUESTED** [review]

File: src/api/entities/confidential/__tests__/ConfidentialTransaction/index.ts (hunk 2 above)
Comment (polymath-eric) [reply src/api/entities/confidential/__tests__/ConfidentialTransaction/index.ts:188]: data doesn't get compared in the error comparision, just code and message.

**Pull request merged**